pkg/asset/tls.go

```diff
@@ -148,14 +148,78 @@ func newEtcdTLSAssets(etcdCACert, etcdClientCert *x509.Certificate, etcdClientKe
 	return assets, nil
 }
 
+func newSelfHostedEtcdTLSAssets(etcdSvcIP, bootEtcdSvcIP string, caCert *x509.Certificate, caPrivKey *rsa.PrivateKey) (Assets, error) {
```

Comment:
So, this method does not allow any user specifying of etcd certs, which is fine, but maybe just add a comment or something.